modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java

```diff
@@ -182,15 +188,37 @@ protected GridCoverage execute() {
 				// merges
 				//
 				// //
-				if (op == null)
+				if (op == null){
 					op = currentSourceCoverage.getRenderedImage();
+					w.setImage(op);
```

Comment:
Might be wrong, but believe I've seen this code a few times... wouldn't it make sense to factor it out, maybe (just ne of the many options) having a ImageWorker.setCoverage(...) that does all this dance?